common/src/main/java/net/william278/husksync/database/mongo/MongoConnectionHandler.java

```diff
@@ -0,0 +1,49 @@
+package net.william278.husksync.database.mongo;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+
+@Getter
+public class MongoConnectionHandler {
+    private final MongoClient mongoClient;
+    private final MongoDatabase database;
+
+    /**
+     * Initiate a connection to a Mongo Server
+     * @param host The IP/Host Name of the Mongo Server
+     * @param port The Port of the Mongo Server
+     * @param username The Username of the user with the appropriate permissions
+     * @param password The Password of the user with the appropriate permissions
+     * @param databaseName The database to use.
+     * @param authDb The database to authenticate with.
+     */
+    public MongoConnectionHandler(@NotNull String host, @NotNull Integer port, @NotNull String username, @NotNull String password, @NotNull  String databaseName, @NotNull String authDb) {
```

Comment:
You could use the primitive type for port.

Reply:
you cant put notnull on primitive

Reply:
that's not a problem. why should it be? It would just give 0 if it's not initialized, and then it would give you errors based on the logic port.